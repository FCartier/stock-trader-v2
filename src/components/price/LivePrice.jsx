import React from "react";

const dynamicDiv = price => {
  let text = '';

  price > 0 
  ? text = "text-green"
  : text = "text-orange";

  return text;
}

const LivePrice = ({price}) => {
  const { lastSalePrice } = price;
  console.log(lastSalePrice)
  return Object.keys(price).length ? (
    <div>

        <div className="price">
            <sup>$</sup>{lastSalePrice}
        </div>

    </div>
  ) : null
};

export default LivePrice;