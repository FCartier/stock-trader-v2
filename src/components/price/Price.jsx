import React from "react";

const Price = ({price}) => {
   return renderCondition(price) ? (
    <div>
        <div className="price">
            <sup>$</sup>{price}
        </div>
    </div>
   ) : null
};


function renderCondition(price){
  if (price === 0) {
    return false
  } else {
    return true
  }
}


export default Price;