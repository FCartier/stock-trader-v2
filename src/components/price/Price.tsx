import * as React from "react";

export interface IPriceProps {
  price: {
    close: number;
    change: number;
    changePercent: number;
    lastSalePrice: number;
    askPrice: number;
  };
}

const Price: React.SFC<IPriceProps> = ({ price }) => {
  return price.close ? (
    <div>
      <div className="price">
        <sup>$</sup>
        {price.close.toFixed(2)}{" "}
        <div className={dynamicDiv(price.change)}>
          {price.change} | {price.changePercent}
          <sup>%</sup>
        </div>
      </div>
    </div>
  ) : (
    <div className="price">
      
      {price.askPrice ? (
        <div>
          <sup>$</sup>
          {price.askPrice.toFixed(2)}
        </div>
      ) : null} 
    </div>
  );
};

const dynamicDiv = (price: number) => {
  return price > 0 ? "text-green" : "text-orange";
};


export default Price;
