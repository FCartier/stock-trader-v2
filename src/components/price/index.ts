import PriceContainer from "./priceContainer"

export default PriceContainer