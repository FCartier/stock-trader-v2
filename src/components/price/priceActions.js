export const FETCH_PRICE_SUCCESS = "FETCH_PRICE_SUCCESS";
export const RECEIVED_NEW_PRICE = 'RECEIVED_NEW_PRICE'

export const priceReceived = payload => ({
    type: RECEIVED_NEW_PRICE,
    payload
  });
  
