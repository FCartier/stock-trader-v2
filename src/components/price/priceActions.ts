import { Action, ActionCreator } from "redux";

export const FETCH_STATIC_PRICE = "FETCH_STATIC_PRICE";
export type FETCH_STATIC_PRICE = typeof FETCH_STATIC_PRICE;

export const RECEIVED_NEW_PRICE = "RECEIVED_NEW_PRICE";
export type RECEIVED_NEW_PRICE = typeof RECEIVED_NEW_PRICE;

export interface StaticPriceAction extends Action {
  type: typeof FETCH_STATIC_PRICE
  payload: string
};

export interface PriceAction extends Action {
  type: typeof RECEIVED_NEW_PRICE;
  payload: string;
}

export const priceReceived: ActionCreator<PriceAction> = payload => ({
  type: RECEIVED_NEW_PRICE,
  payload
});

export const fetchStaticPrice: ActionCreator<StaticPriceAction> = payload => ({
  type: FETCH_STATIC_PRICE,
  payload
});
