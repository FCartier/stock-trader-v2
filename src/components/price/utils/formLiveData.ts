import { IPrice, IKeyStats } from "../types";

export const formLiveData = (price: IPrice, keystats: IKeyStats) => {
    if (price.askPrice && keystats.previousClose) {
      return {
        lastSalePrice: price.lastSalePrice,
        change: (price.lastSalePrice - keystats.previousClose).toFixed(3),
        changePercent:
          ((price.lastSalePrice - keystats.previousClose) / keystats.previousClose).toFixed(3)
      };
    } else if (keystats.previousClose && !price.askPrice) {
      console.log("static triggered")
      return {
        lastSalePrice: keystats.previousClose,
        change: keystats.change,
        changePercent: keystats.changePercent
      };
    } else {
      return null;
    }
  };
