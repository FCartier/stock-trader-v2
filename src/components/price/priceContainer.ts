import Price from "./Price";
import { connect } from "react-redux";
import { GlobalState } from "../../StoreTypes";

const mapStateToProps = ({ price, keystats: { keystats } }: GlobalState) => ({
  price,
  keystats
});

export default connect(
  mapStateToProps,
  null
)(Price);
