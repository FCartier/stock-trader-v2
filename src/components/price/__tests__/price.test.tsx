import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import Price from "../Price";
import { data } from "../__data__/price.data";
import * as formLiveData  from "../utils/formLiveData";

configure({ adapter: new Adapter() });

describe("price", () => {
  test("should render properly with data", () => {
    const component = shallow(<Price {...data} />);
    expect(component).toMatchSnapshot();
  });

  test("should render properly with no data", () => {
    jest.spyOn(formLiveData, "default").mockReturnValue(null)
    const component = shallow(<Price {...data} />);
    expect(component).toMatchSnapshot();
  })
});
