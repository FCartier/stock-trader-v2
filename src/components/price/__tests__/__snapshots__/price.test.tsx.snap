// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`price should render properly with data 1`] = `
<Styled(div)>
  <Styled(div)>
    <Styled(sup)>
      $
    </Styled(sup)>
    221.19
  </Styled(div)>
  <Styled(div)
    className="css-1dt5uw9"
  >
    0.000
     | 
    0.000
    <sup>
      %
    </sup>
  </Styled(div)>
</Styled(div)>
`;

exports[`price should render properly with no data 1`] = `""`;
